nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/PutDatabaseRecordIT.java

```diff
@@ -0,0 +1,312 @@
+        postgres = new PostgreSQLContainer<>("postgres:9.6.12")
+            .withInitScript("PutDatabaseRecordIT/create-person-table.sql");
+        postgres.start();
+    }
+
+    @AfterAll
+    public static void cleanup() {
+        if (postgres != null) {
+            postgres.close();
+            postgres = null;
+        }
+    }
+
+    @BeforeEach
+    public void setup() throws InitializationException, SQLException {
+        truncateTable();
+
+        runner = TestRunners.newTestRunner(PutDatabaseRecord.class);
+        final DBCPConnectionPool connectionPool = new DBCPConnectionPool();
+        runner.addControllerService("connectionPool", connectionPool);
+        runner.setProperty(connectionPool, DBCPProperties.DATABASE_URL, postgres.getJdbcUrl());
+        runner.setProperty(connectionPool, DBCPProperties.DB_USER, postgres.getUsername());
+        runner.setProperty(connectionPool, DBCPProperties.DB_PASSWORD, postgres.getPassword());
+        runner.setProperty(connectionPool, DBCPProperties.DB_DRIVERNAME, postgres.getDriverClassName());
+        runner.enableControllerService(connectionPool);
+
+        final JsonTreeReader jsonReader = new JsonTreeReader();
+        runner.addControllerService("json-reader", jsonReader);
+        runner.setProperty(jsonReader, DateTimeUtils.DATE_FORMAT, "yyyy-MM-dd");
+        runner.setProperty(jsonReader, DateTimeUtils.TIMESTAMP_FORMAT, "yyyy-MM-dd HH:mm:ss.SSSSSS");
+        runner.enableControllerService(jsonReader);
+
+        runner.setProperty(PutDatabaseRecord.RECORD_READER_FACTORY, "json-reader");
+        runner.setProperty(PutDatabaseRecord.DBCP_SERVICE, "connectionPool");
+
+        runner.setProperty(PutDatabaseRecord.TABLE_NAME, "person");
+        runner.setProperty(PutDatabaseRecord.DB_TYPE, "PostgreSQL");
+        runner.setProperty(PutDatabaseRecord.STATEMENT_TYPE, "INSERT");
+    }
+
+
+    @Test
+    public void testSimplePut() throws SQLException {
+        runner.enqueue("""
+            {
+              "name": "John Doe",
+              "age": 50,
+              "favorite_color": "blue"
+            }
+            """);
+        runner.run();
+        runner.assertAllFlowFilesTransferred(PutDatabaseRecord.REL_SUCCESS, 1);
+
+        final Map<String, Object> results = getResults();
+        assertEquals("John Doe", results.get("name"));
+        assertEquals(50, results.get("age"));
+        assertEquals("blue", results.get("favorite_color"));
+    }
+
+    @Test
+    public void testWithDate() throws SQLException {
+        runner.enqueue("""
+            {
+              "name": "John Doe",
+              "age": 50,
+              "dob": "1975-01-01"
+            }
+            """);
+        runner.run();
+        runner.assertAllFlowFilesTransferred(PutDatabaseRecord.REL_SUCCESS, 1);
+
+
+        final Map<String, Object> results = getResults();
+        assertEquals("John Doe", results.get("name"));
+        assertEquals(50, results.get("age"));
```

Comment:
These comparisons are repeated in most of the test methods, along with the same data. It looks like the test methods could be refactored so that the shared behavior is not repeated. That would make this test more maintainable, avoid some of the duplication, and highlight what is unique about the particular test method (Date, Timestamp, Milliseconds, Microseconds, etc).